### PR TITLE
Add first set of MIDS 2 terms

### PR DESCRIPTION
**Pull request by samleeflang**
Adds the first set of MIDS 2 terms to the harmonized datamodel
- [Collector](https://github.com/tdwg/mids/issues/22)
- [DateCollected](https://github.com/tdwg/mids/issues/24)
- [CollectingNumber](https://github.com/tdwg/mids/issues/23)
- [TypeStatus](https://github.com/tdwg/mids/issues/26)

### CODE REVIEW

**Comment by samleeflang** [comment]
Will ignore the Sonar test as the duplication is already minimized by adding all the logic in the parent. However the functions still need to be called that is what is duplicate. Don't see another way (yet) without losing the ability to create custom functions for the retrieval.

File: src/main/java/eu/dissco/core/translator/terms/specimen/Collector.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+  @Override
+  public String retrieveFromDWCA(ArchiveFile archiveFile, Record rec) {
+    return super.searchDWCAForTerm(archiveFile, rec, dwcaTerms);
+  }
+
+  @Override
+  public String retrieveFromABCD(JsonNode unit) {
+    StringBuilder builder = new StringBuilder();
+    combinePossibleValues(unit, builder);
```
Comment (southeo) [inline src/main/java/eu/dissco/core/translator/terms/specimen/Collector.java:26]: Is the result of combinePossibleValues being saved?

**Review by TomDijkema — APPROVED** [review]
Nice! Will be nice to have the collector information so we can display some valid information in the specimen overview of the front-end

File: src/test/java/eu/dissco/core/translator/terms/specimen/CollectorTest.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  private ArchiveFile archiveFile;
+  @Mock
+  private Record rec;
+  private Collector collector;
+
+  @BeforeEach
+  void setup() {
+    collector = new Collector();
+  }
+
```
Comment (southeo) [inline src/test/java/eu/dissco/core/translator/terms/specimen/CollectorTest.java:32]: Could you put this in the declaration and make it final?

**Review by southeo — APPROVED** [review]
:+1:

File: src/test/java/eu/dissco/core/translator/terms/specimen/CollectorTest.java (hunk 2 above)
Comment (samleeflang) [reply src/test/java/eu/dissco/core/translator/terms/specimen/CollectorTest.java:32]: Yes thanks, changed it.

**Pull request merged**